### DISCUSSION TITLE
OR-TOOLS - Pick-up and Delivery Load and Unload Times

### DISCUSSION THREAD

**Discussion by Roosjeva** [google/or-tools#3040] [Routing (and legacy CP) questions]
Python, Trying to solve PDPTW, would like to obtain solution with load and unload time 

I'm trying to add load and unload times.  
I've found to add these load and unload times at the depot.  As demonstrated below. 
But I'm solving a Pick- up and Delivery Problem and I want to add load and unload time at every node. 
Besides the load and unload times depend for every node. 

So how to add this interval for every node and not only on the depot. 

```python
    solver = routing.solver()
    intervals = []
    for i in range(data['num_vehicles']):
        # Add time windows at start of routes
        intervals.append(
            solver.FixedDurationIntervalVar(
                time_dimension.CumulVar(routing.Start(i)),
                data['vehicle_load_time'], 'depot_interval'))
        # Add time windows at end of routes.
        intervals.append(
            solver.FixedDurationIntervalVar(
                time_dimension.CumulVar(routing.End(i)),
                data['vehicle_unload_time'], 'depot_interval'))
```

**Comment by Mizux** [comment]
Why not adding the load/unload time directly to the transit callback ?

**Reply by Roosjeva** [reply]
Transit callback is based on the time to travel right? Or do you mean something else?

when i add additional time to the travel time it means that i don't achieve my time-windows, so that is not an option

**Reply by lperron** [reply]
you adjust the time window.

**Reply by Roosjeva** [reply]
That is not possible 

I've a setted time window 
I need to arrive in that time window but the loading and unloading can take longer. 

So how can I add load and unload times to a pick-up and delivery problem

**Reply by frozar** [reply]
Is there any progress on this topic?

**Reply by th1460** [reply]
This is an important question 
>I need to arrive in that time window but the loading and unloading can take longer

I am trying solve this problem too. Any solution?

**Comment by jmarca** [comment]
I for one don’t know how to answer this question. From the words, it just sounds like asking for the impossible…make a delivery with a long service time fit in a time window that is too small. That must not be the case. You must be asking something else. If so please ask your question with numbers and a clear example of what the problem is.Regards, JamesOn Nov 18, 2022, at 11:11, Thiago Pires ***@***.***> wrote:﻿
This is an important question

I need to arrive in that time window but the loading and unloading can take longer
I am trying solve this problem too. Any solution?
